Replace .format with f-strings

eumiro opened the pull request:
Replace most of the `.format` calls with proper f-strings. Left `.format` where backslashes or complex escapes were used.

The second commit shortens the line using explicit `textwrap.shorten` instead of trivial string slicing. This may deliver different results because it cuts off whole words and not individual characters.  

Fixes #2012

jugmac00 reviewed (changes requested):
@eumiro Thank you for the PR.

Please have a look at our developer documentation. You need to create a changelog entry.

https://tox.readthedocs.io/en/rewrite/development.html#changelog-entries

gaborbernat commented on src/tox/util/pep517/backend.py:
```diff
@@ -23,11 +23,11 @@ def __init__(self, backend_module, backend_obj):
         on_object = self if name.startswith("_") else self.backend
         if not hasattr(on_object, name):
-            raise MissingCommand("{!r} has no attribute {!r}".format(on_object, name))
+            raise MissingCommand(f"{on_object!r} has no attribute {name!r}")
```
The backend needs to run under python 2 too.

gaborbernat commented on tests/demo_pkg_inline/build.py:
```diff
@@ -8,17 +8,17 @@
-wheel = "{}/WHEEL".format(dist_info)
-record = "{}/RECORD".format(dist_info)
+dist_info = f"{name}-{version}.dist-info"
+logic = f"{name}/__init__.py"
```
This file needs to run under python 2 too.

gaborbernat commented on tests/execute/local_subprocess/test_local_subprocess.py:
```diff
@@ -105,14 +105,12 @@ def test_local_execute_write_a_lot(os_env: Dict[str, str]) -> None:
-                "print('a' * {0}, file=sys.stderr);"
-                "print('b' * {0}, file=sys.stdout);"
-            ).format(count),
+            f"import sys; import time; from datetime import datetime; import os;"
```
This too I think 🤔

eumiro replied:
Looks like this file could be modernized to Python 3 as it was not mentioned in #2015 ? :thinking:

gaborbernat replied:
It does look like 🤔 unless some tests fail in which case we need to exclude it from pyupgrade 👍🏻 But I think it's alright.

eumiro commented:
> Some files still need to be python 2 compatible, to allow and test for running python 2 envs from tox.

What would be the best way to tag these files compatible with both 2/3? Something in the header?

gaborbernat commented:
That can be, currently they're tagged in my head and hopefully by the test suite.

eumiro commented:
Reverted changes in Py2/3 files, added changelog entry.

And, @gaborbernat, I'd like to find a standard way of marking files to be Py2/3 compatible to avoid contributors from keeping fixing the Py2 relics from them.

gaborbernat commented:
> Reverted changes in Py2/3 files, added changelog entry.
> 
> And, @gaborbernat, I'd like to find a standard way of marking files to be Py2/3 compatible to avoid contributors from keeping fixing the Py2 relics from them.

Can't think of any better than mention this in the module docstring...

eumiro commented:
There's one formatting test failing, can have a look at it later.

And created #2013 to discuss the Py2/3 problem. Module docstring is a good idea.

eumiro commented:
The failed tests come from the usage of `textwrap.shorten`. Now the question is, how do you want to handle the words in the output? Should it cut after X characters like before or should it show whole words and possibly cut earlier?

gaborbernat commented:
> The failed tests come from the usage of `textwrap.shorten`. Now the question is, how do you want to handle the words in the output? Should it cut after X characters like before or should it show whole words and possibly cut earlier?

As long as the ... are added I'm happy to cut it at the word boundary.

gaborbernat commented:
@eumiro in light of your recent PRs I've invited you to the maintainers' group 👏🏻 No pressure though, you can keep contributing as much or as little you want, this just gives you some more permissions to review/approve/triage. Thanks for all your very helpful feedback and effort to the project 🙇🏻.

eumiro commented:
Thank you for your active feedback, @gaborbernat and @jugmac00, and also for the opportunity to take part in the future of tox4!
